packages/express-did-auth/test/request-counter.test.ts

```diff
@@ -0,0 +1,63 @@
+import { ErrorCodes } from '../src/errors'
+import RequestCounter from '../src/classes/request-counter'
+import MockDate from 'mockdate'
+
+describe('RequestCounter', () => {
+  afterEach(() => MockDate.reset())
+
+  describe('count', () => {
+    const did = 'did:ethr:rsk:testnet:0xd69ced736454347be68aead53fcc1678cb9a70ef'
+
+    it('should throw an error if no did', () => {
+      const counter = new RequestCounter({})
+
+      expect(() => counter.count(undefined)).toThrow(ErrorCodes.INVALID_DID)
+    })
+
+    it('should count for the first time with a valid did', () => {
+      const counter = new RequestCounter({})
+
+      counter.count(did)
+    })
+
+    it('should not allow to count more than the limit', () => {
+      const counter = new RequestCounter({ maxRequestsPerTimeSlot: 5, timeSlotInSeconds: 100 })
+
+      for (let i = 0; i < 5; i++) counter.count(did)
+
+      // 6th time should throw an error
+      expect(() => counter.count(did)).toThrow(ErrorCodes.MAX_REQUESTS_REACHED)
+    })
+
+    it('should allow to count more than the limit if the requests are divided in different timeslots', () => {
```

Comment:
I don't understand the description. And, I would add
```
expect(() => counter.count(did)).toThrow(ErrorCodes.MAX_REQUESTS_REACHED)
```
after the given timeslot exceeded.

Then, why does he need to wait `150`?

Reply:
What is intended to test here is that the counter is "reset" after the timeslot and it does not fail, I've added that `150` as a minimum grace period

Reply:
This test checks if the counter is "reset" after the timeslot is elapsed, because it should allow a user to submit more requests after it. In this case, I'm expecting it to do not throw the `MAX_REQUESTS_REACHED` exception